### ISSUE TITLE
NuGetDefense : OSS Index scan failed

### ISSUE THREAD

**Issue by MatheusXavier** [digitalcoyote/NuGetDefense#30]
Hello, I'm getting this error when running unit tests inside a docker container:

"IdentityProvider.csproj : error : NuGetDefense : OSS Index scan failed with exception: System.AggregateException: One or more errors occurred. (The input does not contain any JSON tokens. Expected the input to start with a valid JSON token, when isFinalBlock is true. Path: $ | LineNumber: 0 | BytePositionInLine: 0.)"

When I run on my machine it works, but when I run on my pipeline I got this error, I'm not sure if it's because the docker.

**Comment by digitalcoyote** [comment]
This is very similar to #28 and is likely caused by not getting a json response from OSSIndex. This could be caused by local network issues. I have plans to check the response and provide better error messages in the future, but I may not be able to start on that til next month (in the process of moving and I'll be without internet for a bit between selling our house and buying the new one). In the mean time, you may want to disable OSSIndex in the settings or verify there are no network/firewall issues that could prevent your container from reaching OSSIndex. OSSIndex may block older versions of the package because of #29.

**Comment by MatheusXavier** [comment]
When I'm using the version 1.0.6 I can run succesfully inside docker on my local machine, but I got this error when running on Azure CI/CD pipeline, but when I upgraded the package for version 1.0.7.1 I got this error even when I'm running on my local machine inside the docker container. I couldn't find any trouble with the network inside de container, so for now I will disable the OSSIndex as your suggestion.

**Comment by digitalcoyote** [comment]
1.0.7.2 may have fixed this issue. It appears to have corrected #28

**Comment by ekarcnevets** [comment]
I am on 1.0.7.2 and intermittently still get this error. It seems to go away overnight but does block builds until it self-resolves which isn't ideal :(

**Comment by digitalcoyote** [comment]
@ekarcnevets I believe you are probably running into problems with OSSIndex's rate limiting. In later versions of NuGetDefense, I added an explicit warning. I'm working on a few methods to allow caching responses, but I would recommend for now that you alter the NuGetDefense.json to set BreakIfCannotRun to false for OSSIndex.

Beyond that, I'd recommend upgrading beyond 1.0.7.2 as there have been multiple other issues with the OSSIndex scanner fixed since that version. In addition, I added optional OSSIndex credentials that extend the rate limits slightly. Any 1.x version will still use the dotnetcore3.1 sdk.
